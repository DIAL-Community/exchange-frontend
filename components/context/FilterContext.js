import { createContext, useEffect, useState } from 'react'
import { useRouter } from 'next/router'

const FilterContext = createContext()
const FilterDispatchContext = createContext()

const FilterProvider = ({ children }) => {
  const [search, setSearch] = useState('')
  // Task tracker context only
  const [showFailedOnly, setShowFailedOnly] = useState(false)
  const [showGovStackOnly, setShowGovStackOnly] = useState(false)
  const [showMature, setShowMature] = useState(false)

  const [categoryTypes, setCategoryTypes] = useState([])
  const [sdgs, setSdgs] = useState([])
  const [useCases, setUseCases] = useState([])
  const [workflows, setWorkflows] = useState([])

  const [origins, setOrigins] = useState([])
  const [countries, setCountries] = useState([])
  const [sectors, setSectors] = useState([])
  const [organizations, setOrganizations] = useState([])
  const [tags, setTags] = useState([])
  const [datasetTypes, setDatasetTypes] = useState([])

  const [aggregators, setAggregators] = useState([])
  const [operators, setOperators] = useState([])
  const [services, setServices] = useState([])

  const [years, setYears] = useState([])

  const [products, setProducts] = useState([])

  const [buildingBlocks, setBuildingBlocks] = useState([])
  const [showClosed, setShowClosed] = useState(false)

  const [aggregator, setAggregator] = useState(false)
  const [endorser, setEndorser] = useState(false)
  const [endorserLevel, setEndorserLevel] = useState('')

  const [specialties, setSpecialties] = useState([])
  const [certifications, setCertifications] = useState([])

  const [isEndorsed, setIsEndorsed] = useState(false)
  const [endorsers, setEndorsers] = useState([])
  const [licenseTypes, setLicenseTypes] = useState([])
  const [isLinkedWithDpi, setIsLinkedWithDpi] = useState(false)
  const [showDpgaOnly, setShowDpgaOnly] = useState(false)

  const [softwareCategories, setSoftwareCategories] = useState([])
  const [softwareFeatures, setSoftwareFeatures] = useState([])
  const [comparedProducts, setComparedProducts] = useState([])
  const [productStage, setProductStage] = useState(null)

  const [showBeta, setShowBeta] = useState(false)

  const router = useRouter()
  useEffect(() => {
    const handleRouteChange = (url) => {
      // Transitioning to other pages (pathname changing).
      if (url.indexOf(router.pathname) < 0) {
        setSearch('')
        setShowFailedOnly(false)
        setShowGovStackOnly(false)
        setShowMature(false)

        setCategoryTypes([])
        setSdgs([])
        setUseCases([])
        setWorkflows([])

        setOrigins([])
        setCountries([])
        setSectors([])
        setOrganizations([])
        setTags([])
        setDatasetTypes([])

        setAggregators([])
        setOperators([])
        setServices([])

        setYears([])

        setProducts([])

        setBuildingBlocks([])
        setShowClosed(false)

        setAggregator(false)
        setEndorser(false)
        setEndorserLevel('')

        setSpecialties([])
        setCertifications([])

        setIsEndorsed(false)
        setEndorsers([])
        setLicenseTypes([])
        setIsLinkedWithDpi(false)
        setShowDpgaOnly(false)

        setSoftwareCategories([])
        setSoftwareFeatures([])
        setComparedProducts([])
        setProductStage('')

        setShowBeta(false)
      }
    }

    router.events.on('routeChangeComplete', handleRouteChange)

    return () => {
      router.events.off('routeChangeComplete', handleRouteChange)
    }
  }, [router.events, router.asPath, router.pathname])

  const valueProps = {
    search,

    showFailedOnly,
    showGovStackOnly,
    showMature,
    categoryTypes,
    sdgs,
    useCases,
    workflows,

    countries,
    datasetTypes,
    organizations,
    origins,
    sectors,
    tags,

    aggregators,
    operators,
    services,

    years,

    products,

    buildingBlocks,
    showClosed,

    aggregator,
    endorser,
    endorserLevel,
    specialties,
    certifications,

    isEndorsed,
    endorsers,
    licenseTypes,
    isLinkedWithDpi,
    showDpgaOnly,
    softwareCategories,
    softwareFeatures,
    comparedProducts,
    productStage,

    showBeta
  }

  const dispatchProps = {
    setSearch,
    setShowFailedOnly,
    setShowGovStackOnly,
    setShowMature,
    setCategoryTypes,
    setSdgs,
    setUseCases,
    setWorkflows,

    setCountries,
    setDatasetTypes,
    setOrganizations,
    setOrigins,
    setSectors,
    setTags,

    setAggregators,
    setOperators,
    setServices,

    setYears,

    setProducts,

    setBuildingBlocks,
    setShowClosed,

    setAggregator,
    setEndorser,
    setEndorserLevel,
    setSpecialties,
    setCertifications,

    setIsEndorsed,
    setEndorsers,
    setLicenseTypes,
    setIsLinkedWithDpi,
    setShowDpgaOnly,
    setSoftwareCategories,
    setSoftwareFeatures,
    setComparedProducts,
    setProductStage,

    setShowBeta
  }

  return (
    <FilterContext.Provider value={{ ...valueProps }}>
      <FilterDispatchContext.Provider value={{ ...dispatchProps }}>
        {children}
      </FilterDispatchContext.Provider>
    </FilterContext.Provider>
  )
}

export { FilterProvider, FilterContext, FilterDispatchContext }
