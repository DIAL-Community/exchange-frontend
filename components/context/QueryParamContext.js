import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
