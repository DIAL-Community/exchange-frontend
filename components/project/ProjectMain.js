import ProjectMainLeft from './ProjectMainLeft'
import ProjectMainRight from './ProjectMainRight'

const ProjectMain = ({ activeTab }) => {
  return (
    <div className='px-4 lg:px-8 xl:px-56'>
      <div className='grid grid-cols-4 gap-x-8'>
        <div className='hidden md:block col-span-1'>
          <ProjectMainLeft activeTab={activeTab} />
        </div>
        <div className='col-span-4 md:col-span-3'>
          <ProjectMainRight activeTab={activeTab} />
        </div>
      </div>
    </div>
  )
}

export default ProjectMain
