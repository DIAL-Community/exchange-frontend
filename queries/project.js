import { gql } from '@apollo/client'

export const PROJECT_SEARCH_QUERY = gql`
  query Projects($search: String!) {
    projects(search: $search) {
      id
      name
      slug
      origin {
        slug
      }
    }
  }
`

export const PROJECT_QUERY = gql`
  query Project($slug: String!) {
    project(slug: $slug) {
      id
      name
      slug
      startDate
      endDate
      projectUrl
      projectDescription {
        description
        locale
      }
      products {
        slug
      }
      organizations {
        slug
      }
    }
  }
`
