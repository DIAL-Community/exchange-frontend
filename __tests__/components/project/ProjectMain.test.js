import { screen } from '@testing-library/dom'
import { FilterProvider } from '../../../components/context/FilterContext'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import ProjectMain from '../../../components/project/ProjectMain'
import { PAGINATED_PROJECTS_QUERY, PROJECT_PAGINATION_ATTRIBUTES_QUERY } from '../../../components/shared/query/project'
import { render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { mockNextUseRouter, mockTenantApi } from '../../utils/nextMockImplementation'
import { paginatedProjects, projectPaginationAttribute } from './data/ProjectMain.data'

mockTenantApi()
mockNextUseRouter()
describe('Unit tests for the project main page.', () => {
  test('Should render list of projects.', async () => {
    const mockProjectPaginationAttribute = generateMockApolloData(
      PROJECT_PAGINATION_ATTRIBUTES_QUERY,
      {
        search: '',
        countries: [],
        products: [],
        organizations: [],
        sectors: [],
        tags: [],
        sdgs: [],
        origins: []
      },
      null,
      projectPaginationAttribute
    )
    const mockPaginatedProjects = generateMockApolloData(
      PAGINATED_PROJECTS_QUERY,
      {
        search: '',
        countries: [],
        products: [],
        organizations: [],
        sectors: [],
        tags: [],
        sdgs: [],
        origins: [],
        limit: 12,
        offset: 0
      },
      null,
      paginatedProjects
    )

    const { container } = render(
      <CustomMockedProvider mocks={[mockPaginatedProjects, mockProjectPaginationAttribute]}>
        <QueryParamContextProvider>
          <FilterProvider>
            <ProjectMain activeTab={0} />
          </FilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('HelloMAMA')).toBeInTheDocument()
    expect(await screen.findByText('Hoji Mobile Data Collection and Analysis Platform')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
