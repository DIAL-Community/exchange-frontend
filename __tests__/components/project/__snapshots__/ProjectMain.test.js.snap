// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the project main page. Should render list of projects. 1`] = `
<div>
  <section
    aria-atomic="false"
    aria-label="Notifications Alt+T"
    aria-live="polite"
    aria-relevant="additions text"
    class="Toastify"
  />
  <div
    class="px-4 lg:px-8 xl:px-24 3xl:px-56"
  >
    <div
      class="grid grid-cols-4 gap-x-8"
    >
      <div
        class="hidden md:block col-span-1"
      >
        <div
          class="bg-dial-slate-100 h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="flex flex-col gap-y-2"
              >
                <div
                  class="text-sm font-semibold text-dial-sapphire"
                >
                  Filter your results by
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Product
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Organization
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Sector
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
                <div
                  class="flex flex-col gap-y-3"
                >
                  <a
                    class="flex"
                    href="#"
                  >
                    <div
                      class="text-dial-stratos text-sm py-2"
                    >
                      Tag
                    </div>
                    <svg
                      class="ml-auto text-dial-stratos my-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
                      />
                    </svg>
                  </a>
                </div>
                <hr
                  class="border-b border-dial-slate-200"
                />
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <div
                  class="text-sm font-semibold text-dial-sapphire"
                >
                  <button
                    type="button"
                  >
                    <div
                      class="flex text-dial-sapphire gap-3"
                    >
                      <div
                        class="my-auto"
                      >
                        Refine with more filters
                      </div>
                      <div
                        class="ml-auto py-2 text-xl"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 448 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                          />
                        </svg>
                      </div>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col gap-3 py-3"
            >
              <div
                class="text-dial-sapphire font-semibold"
              >
                Bookmark
              </div>
              <button
                class="group flex flex-row gap-x-3"
                type="button"
              >
                <img
                  alt="Logo of Bookmark."
                  class="object-contain"
                  height="40"
                  src="/ui/v1/bookmark-icon.svg"
                  width="40"
                />
                <div
                  class="my-auto border-b border-transparent group-hover:border-dial-slate-500"
                >
                  <div
                    class="text-sm text-dial-stratos text-left"
                  >
                    Bookmark this page.
                  </div>
                </div>
              </button>
              <div
                class="text-sm text-dial-stratos"
              >
                Your bookmarks can be found in your Account page.
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
            <div
              class="flex flex-col gap-3 py-3"
            >
              <div
                class="text-dial-sapphire font-semibold"
              >
                Share
              </div>
              <div
                class="flex flex-row gap-3"
              >
                <button>
                  <img
                    alt="Logo of LinkedIn."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/linkedin-icon.svg"
                    width="40"
                  />
                </button>
                <button>
                  <div
                    class="bg-dial-iris-blue rounded-full w-[40px] h-[40px]"
                  >
                    <div
                      class="flex items-center h-full"
                    >
                      <img
                        alt="Logo of WhatsApp."
                        class="my-auto mx-auto"
                        height="20"
                        src="/ui/v1/whatsapp-icon.svg"
                        width="20"
                      />
                    </div>
                  </div>
                </button>
                <button>
                  <img
                    alt="Logo of Email."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/email-icon.svg"
                    width="40"
                  />
                </button>
                <button>
                  <img
                    alt="Logo of Copy."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/copy-icon.svg"
                    width="40"
                  />
                </button>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="col-span-4 md:col-span-3"
      >
        <div
          class="min-h-[50vh]"
        >
          <div
            class="py-3"
          >
            <div
              class="flex flex-row items-center justify-center gap-1 relative"
            >
              <div
                class="block md:hidden h-full"
              >
                <button
                  class="mt-[0.5rem]"
                  type="button"
                >
                  <svg
                    class="text-2xl text-dial-slate-500 mx-auto"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13 5h9v2h-9zM2 7h7v2h2V3H9v2H2zm7 10h13v2H9zm10-6h3v2h-3zm-2 4V9.012h-2V11H2v2h13v2zM7 21v-6H5v2H2v2h3v2z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="flex flex-row gap-2"
              >
                <div
                  class="text-sm my-auto"
                >
                  View as:
                </div>
                <button
                  class="text-sm flex gap-1 items-center justify-center"
                  disabled=""
                  title="View data as grid"
                >
                  <svg
                    class="text-dial-slate-500"
                    fill="currentColor"
                    height="24"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M240 240H32V32h208zm240 0H272V32h208zM240 480H32V272h208zm240 0H272V272h208z"
                    />
                  </svg>
                </button>
                <div
                  class="border-r"
                />
                <button
                  class="text-sm flex gap-1 items-center justify-center"
                  title="View data as list"
                >
                  <svg
                    class="text-dial-slate-300"
                    fill="currentColor"
                    height="36"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="36"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M144 144h320M144 256h320M144 368h320"
                      fill="none"
                      stroke-linejoin="round"
                      stroke-width="48"
                    />
                    <path
                      d="M64 128h32v32H64zm0 112h32v32H64zm0 112h32v32H64z"
                      fill="none"
                      stroke-linecap="square"
                      stroke-linejoin="round"
                      stroke-width="32"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="ml-auto w-full max-w-prose"
              >
                <label
                  class="relative"
                  for="search-bar"
                >
                  <input
                    class="w-full border text-sm"
                    id="search-bar"
                    placeholder="Search"
                    type="text"
                    value=""
                  />
                  <svg
                    class="absolute top-1/2 transform -translate-y-1/2 right-2"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </label>
              </div>
            </div>
          </div>
          <div
            class="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-4"
          >
            <div>
              <div
                class="relative"
              >
                <a
                  href="/projects/hellomama"
                >
                  <div
                    class="cursor-pointer hover:rounded-lg hover:shadow-lg"
                  >
                    <div
                      class="bg-white border shadow-lg rounded-xl h-[21rem]"
                    >
                      <div
                        class="flex flex-col h-full"
                      >
                        <div
                          class="flex justify-center items-center bg-white rounded-xl border-4 border-dial-violet py-10 mx-4 my-4 max-h-[12rem]"
                        >
                          <div
                            class="flex flex-row gap-4 justify-center items-center"
                          >
                            <img
                              alt="Logo of Product."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="/ui/v1/product-header.svg"
                            />
                            <img
                              alt="Logo of Organization."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/organizations/jhpiego.png"
                            />
                          </div>
                        </div>
                        <div
                          class="px-6 text-xl text-center font-semibold line-clamp-1"
                        >
                          HelloMAMA
                        </div>
                        <div
                          class="px-6 py-2 text-xs text-dial-stratos font-medium"
                        >
                          <span
                            class="text-center line-clamp-3"
                          >
                            Some parsed description.
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <div
                    class="flex flex-row gap-2"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/projects/hoji_mobile_data_collection_and"
                >
                  <div
                    class="cursor-pointer hover:rounded-lg hover:shadow-lg"
                  >
                    <div
                      class="bg-white border shadow-lg rounded-xl h-[21rem]"
                    >
                      <div
                        class="flex flex-col h-full"
                      >
                        <div
                          class="flex justify-center items-center bg-white rounded-xl border-4 border-dial-violet py-10 mx-4 my-4 max-h-[12rem]"
                        >
                          <div
                            class="flex flex-row gap-4 justify-center items-center"
                          >
                            <img
                              alt="Logo of Product."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="/ui/v1/product-header.svg"
                            />
                            <img
                              alt="Logo of Organization."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="/ui/v1/organization-header.svg"
                            />
                          </div>
                        </div>
                        <div
                          class="px-6 text-xl text-center font-semibold line-clamp-1"
                        >
                          Hoji Mobile Data Collection and Analysis Platform
                        </div>
                        <div
                          class="px-6 py-2 text-xs text-dial-stratos font-medium"
                        >
                          <span
                            class="text-center line-clamp-3"
                          >
                            Some parsed description.
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <div
                    class="flex flex-row gap-2"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/projects/usaid_funded__catss_openmrs_elec"
                >
                  <div
                    class="cursor-pointer hover:rounded-lg hover:shadow-lg"
                  >
                    <div
                      class="bg-white border shadow-lg rounded-xl h-[21rem]"
                    >
                      <div
                        class="flex flex-col h-full"
                      >
                        <div
                          class="flex justify-center items-center bg-white rounded-xl border-4 border-dial-violet py-10 mx-4 my-4 max-h-[12rem]"
                        >
                          <div
                            class="flex flex-row gap-4 justify-center items-center"
                          >
                            <img
                              alt="Logo of Product."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/products/openmrs.png"
                            />
                            <img
                              alt="Logo of Organization."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/organizations/management_sciences_for_health.png"
                            />
                          </div>
                        </div>
                        <div
                          class="px-6 text-xl text-center font-semibold line-clamp-1"
                        >
                          USAID funded  - CaTSS OpenMRS Electronic Medical Record System for 19  ART Clinics
                        </div>
                        <div
                          class="px-6 py-2 text-xs text-dial-stratos font-medium"
                        >
                          <span
                            class="text-center line-clamp-3"
                          >
                            Some parsed description.
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <div
                    class="flex flex-row gap-2"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/projects/mtrac"
                >
                  <div
                    class="cursor-pointer hover:rounded-lg hover:shadow-lg"
                  >
                    <div
                      class="bg-white border shadow-lg rounded-xl h-[21rem]"
                    >
                      <div
                        class="flex flex-col h-full"
                      >
                        <div
                          class="flex justify-center items-center bg-white rounded-xl border-4 border-dial-violet py-10 mx-4 my-4 max-h-[12rem]"
                        >
                          <div
                            class="flex flex-row gap-4 justify-center items-center"
                          >
                            <img
                              alt="Logo of Product."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/products/rapidpro.png"
                            />
                            <img
                              alt="Logo of Organization."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/organizations/united_nations_childrens_fund_un.png"
                            />
                          </div>
                        </div>
                        <div
                          class="px-6 text-xl text-center font-semibold line-clamp-1"
                        >
                          mTrac
                        </div>
                        <div
                          class="px-6 py-2 text-xs text-dial-stratos font-medium"
                        >
                          <span
                            class="text-center line-clamp-3"
                          >
                            Some parsed description.
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <div
                    class="flex flex-row gap-2"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/projects/electronic_integrated_disease_su"
                >
                  <div
                    class="cursor-pointer hover:rounded-lg hover:shadow-lg"
                  >
                    <div
                      class="bg-white border shadow-lg rounded-xl h-[21rem]"
                    >
                      <div
                        class="flex flex-col h-full"
                      >
                        <div
                          class="flex justify-center items-center bg-white rounded-xl border-4 border-dial-violet py-10 mx-4 my-4 max-h-[12rem]"
                        >
                          <div
                            class="flex flex-row gap-4 justify-center items-center"
                          >
                            <img
                              alt="Logo of Product."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/products/dhis2.png"
                            />
                            <img
                              alt="Logo of Organization."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/organizations/giz.png"
                            />
                          </div>
                        </div>
                        <div
                          class="px-6 text-xl text-center font-semibold line-clamp-1"
                        >
                          electronic Integrated Disease Surveillance and Response (eIDSR)
                        </div>
                        <div
                          class="px-6 py-2 text-xs text-dial-stratos font-medium"
                        >
                          <span
                            class="text-center line-clamp-3"
                          >
                            Some parsed description.
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <div
                    class="flex flex-row gap-2"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/projects/ugandaemr_openmrs"
                >
                  <div
                    class="cursor-pointer hover:rounded-lg hover:shadow-lg"
                  >
                    <div
                      class="bg-white border shadow-lg rounded-xl h-[21rem]"
                    >
                      <div
                        class="flex flex-col h-full"
                      >
                        <div
                          class="flex justify-center items-center bg-white rounded-xl border-4 border-dial-violet py-10 mx-4 my-4 max-h-[12rem]"
                        >
                          <div
                            class="flex flex-row gap-4 justify-center items-center"
                          >
                            <img
                              alt="Logo of Product."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/products/openmrs.png"
                            />
                            <img
                              alt="Logo of Organization."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/organizations/world_health_organization_who.png"
                            />
                          </div>
                        </div>
                        <div
                          class="px-6 text-xl text-center font-semibold line-clamp-1"
                        >
                          UgandaEMR (OpenMRS)
                        </div>
                        <div
                          class="px-6 py-2 text-xs text-dial-stratos font-medium"
                        >
                          <span
                            class="text-center line-clamp-3"
                          >
                            Some parsed description.
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <div
                    class="flex flex-row gap-2"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                class="relative"
              >
                <a
                  href="/projects/improving_health_worker_adherenc"
                >
                  <div
                    class="cursor-pointer hover:rounded-lg hover:shadow-lg"
                  >
                    <div
                      class="bg-white border shadow-lg rounded-xl h-[21rem]"
                    >
                      <div
                        class="flex flex-col h-full"
                      >
                        <div
                          class="flex justify-center items-center bg-white rounded-xl border-4 border-dial-violet py-10 mx-4 my-4 max-h-[12rem]"
                        >
                          <div
                            class="flex flex-row gap-4 justify-center items-center"
                          >
                            <img
                              alt="Logo of Product."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="/ui/v1/product-header.svg"
                            />
                            <img
                              alt="Logo of Organization."
                              class="object-contain max-h-[4rem] w-[4rem]"
                              src="http://localhost:3000//assets/organizations/malaria_consortium.png"
                            />
                          </div>
                        </div>
                        <div
                          class="px-6 text-xl text-center font-semibold line-clamp-1"
                        >
                          Improving health worker adherence to IPTp provision guidelines through sending text messages
                        </div>
                        <div
                          class="px-6 py-2 text-xs text-dial-stratos font-medium"
                        >
                          <span
                            class="text-center line-clamp-3"
                          >
                            Some parsed description.
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </a>
                <div
                  class="absolute top-2 right-2"
                >
                  <div
                    class="flex flex-row gap-2"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-y-4 my-6"
          >
            <hr
              class="border-b border-dial-slate-300"
            />
            <div
              class="flex flex-col xl:flex-row gap-x-16 gap-y-3"
            >
              <div
                class="my-auto text-sm font-semibold"
              >
                Showing 1 - 12 of 4262 total
              </div>
              <ul
                aria-label="Pagination"
                class="flex gap-x-3 text-sm text-dial-slate-300 font-semibold"
                role="navigation"
              >
                <li
                  class="border border-current rounded-md opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Previous page"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500 "
                    rel="prev"
                    role="button"
                    tabindex="-1"
                  >
                    &lt;
                  </a>
                </li>
                <li
                  class="border border-current rounded-md text-dial-iris-blue"
                >
                  <a
                    aria-current="page"
                    aria-label="Page 1 is your current page"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    rel="canonical"
                    role="button"
                    tabindex="-1"
                  >
                    1
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-label="Page 2"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    rel="next"
                    role="button"
                    tabindex="0"
                  >
                    2
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-label="Page 3"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    role="button"
                    tabindex="0"
                  >
                    3
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-label="Page 4"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    role="button"
                    tabindex="0"
                  >
                    4
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-label="Page 5"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    role="button"
                    tabindex="0"
                  >
                    5
                  </a>
                </li>
                <li
                  class="my-auto"
                >
                  <a
                    aria-label="Jump forward"
                    role="button"
                    tabindex="0"
                  >
                    ...
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-label="Page 356"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    role="button"
                    tabindex="0"
                  >
                    356
                  </a>
                </li>
                <li
                  class="border border-current rounded-md"
                >
                  <a
                    aria-disabled="false"
                    aria-label="Next page"
                    class="block w-10 h-10 leading-10 text-center text-dial-slate-500"
                    rel="next"
                    role="button"
                    tabindex="0"
                  >
                    &gt;
                  </a>
                </li>
              </ul>
            </div>
            <hr
              class="border-b border-dial-slate-300"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
